meinberlin/apps/votes/admin.py

```diff
@@ -11,19 +12,27 @@ class VotingTokenAdmin(admin.ModelAdmin):
         "token_hash",
         "module",
         "is_active",
-        "package_number",
+        "package",
         "allowed_votes",
     )
-    readonly_fields = ("token", "token_hash", "package_number")
+    readonly_fields = ("token", "token_hash", "package")
     list_display = ("pk", "__str__", "project", "module", "module_name", "is_active")
     list_filter = ("module__project",)
 
+    def delete_model(self, request, obj):
+        obj.package.size -= 1
+        obj.package.save()
+        super().delete_model(request, obj)
+
     def module_name(self, token):
         return token.module.name
 
     def save_model(self, request, obj, form, change):
-        if obj.package_number is None:
-            obj.package_number = VotingToken.next_package_number(obj.module)
+        if obj.package is None:
```

Comment:
We need ```not hasattr(obj, 'package')``` here, otherwise it throws an error when creating a token..

Reply:
nice catch, fixed